feature/user_profile/presentation/src/main/java/com/sanaa/presentation/screen/myRating/MyRatingScreen.kt

```diff
@@ -55,58 +53,60 @@ fun MyRatingScreen(
     }
 
     var snack by remember { mutableStateOf<SnackData?>(null) }
-    val state = viewModel.state.collectAsStateWithLifecycle()
-
-    LaunchedEffect(Unit) {
-        viewModel.effect.collectLatest { effect ->
-            when (effect) {
-                is MyRatingScreenEffect.NavigateBack -> {
-                    navController.popBackStack()
-                }
+    val state by viewModel.state.collectAsStateWithLifecycle()
 
-                is MyRatingScreenEffect.ShowErrorSnackBar -> {
-                    snack = SnackData(message = failedMsg, isError = true)
+    MyRatingScreenEffectsHandler(
+        effects = viewModel.effect,
+        onNavigateBack = { navController.popBackStack() },
+        onShowErrorSnackBar = { message -> snack = SnackData(message = message, isError = true) },
+        onShowSuccessSnackBar = { message -> snack = SnackData(message = message, isError = false) },
+        onNavigateToMediaDetails = { id, type ->
+            detailsApi.launch(
+                context = navController.context,
+                id = id,
+                startRoute = when (type) {
+                    MediaTypeUi.MOVIE -> StartRoute.MOVIE
+                    MediaTypeUi.TV_SHOW -> StartRoute.SERIES
                 }
-
-                is MyRatingScreenEffect.ShowSuccessSnackBar -> {
-                    snack = SnackData(message = successMsg, isError = false)
-                }
-
-                is MyRatingScreenEffect.NavigateToMediaDetails -> {
-                    detailsApi.launch(
-                        context = navController.context,
-                        id = effect.mediaId,
-                        startRoute = when (effect.mediaTypeUi) {
-                            MediaTypeUi.MOVIE -> StartRoute.MOVIE
-                            MediaTypeUi.TV_SHOW -> StartRoute.SERIES
-                        }
-                    )
-                }
-            }
+            )
         }
-    }
-
+    )
 
-    Box(modifier = Modifier.fillMaxSize()) {
-        MyRatingScreenContent(
-            state = state.value,
-            interactionListener = viewModel
-        )
-        AnimatedSnackBarHost(
-            data = snack,
-            onDismiss = { snack = null },
-        )
-    }
+    MyRatingScreenContent(
+        state = state,
+        interactionListener = viewModel,
+        snack = snack,
+        onDismissSnack = { snack = null }
+    )
 }
 
 @Composable
-fun MyRatingScreenContent(
+private fun MyRatingScreenEffectsHandler(
+    onNavigateBack: () -> Unit,
+    onShowErrorSnackBar: (String) -> Unit,
+    onShowSuccessSnackBar: (String) -> Unit,
+    onNavigateToMediaDetails: (Int, MediaTypeUi) -> Unit,
+    effects:SharedFlow<MyRatingScreenEffect>
+) {
+    LaunchedEffect(Unit) {
+        effects.collectLatest { effect ->
+            when (effect) {
+                is MyRatingScreenEffect.NavigateBack -> onNavigateBack()
```

Comment:
You can import this as a static import, it make the code smaller:
```suggestion
                is NavigateBack -> onNavigateBack()
```